### PR TITLE
Proposal: Move the default theme into its own 'color' stylesheet

### PR DESCRIPTION
**Pull request by jpanther**
In order to simplify the amount of styles being overridden by the browser I'm proposing this PR to split the default theme out into its own `orange.css` stylesheet. This means that you no longer need to include the base stylesheet and then override all the styles when a different colour is selected in the theme config.

The benefit here is that it greatly reduces the browser rendering work and removes the chance of conflict between the orange and other themes. It also minifies the total css used in the theme at runtime as everything is contained in a single file.

### CODE REVIEW

**Comment by jpanther** [comment]
Thanks for the feedback. I probably didn't go into enough detail in my original comment to fully explain this change. Here's a bit more information that hopefully covers your questions.

**TL;DR:** The PR doesn't change the way themes can be overridden but custom styles aren't as simple as your example. There's plenty of opportunity to allow for further customisation and this is just a first step so there's a clean foundation to build customisations on top of.

### Background

The main reason I looked into splitting the default theme out is that currently when you try to make your own palette and override the variables, there's artefacts from the default style that cannot be overridden. If you use the current default orange theme and create a custom `static/style.css` and override `--accent` there are still some orange styles left behind. This is partly due to some of the styles not using the variables (some use `--color-mod` which lose the ability to be overridden with variables) but some get overridden because of the load order of the default=>theme=>custom stylesheets and the custom ones coming too late in the chain to have precedence.

### What this change does

So with that background in mind, what this change essentially does is increase the customisability of the theme palette by moving all the colour specificity into its own stylesheet. The `:root` variables are all still present, they are just now in each of their own stylesheets, and so they can still be overridden in custom palettes. I didn't change the way this is done in the sense that you can still create `static/style.css` and override whatever variables you choose, but you still have to pick one of the built-in theme colours as your base in your `config.toml`.

So you might set `config.toml` to be:
```toml
[params]
  themeColor = "blue"
```

And then in `static/style.css` you can override the same as you can now:
```css
 :root {
   --accent: #22dd33;
 }
```

I've tested and this does what you'd expect, you get the blue base style with bright green accent:
Screen Shot 2019-11-04 at 09 56 33


### Custom themes

This change currently doesn't allow you to make your own theme. So your example where you create `static/tomato.css` and override the styles isn't possible. Although this could be something that could be implemented, I just preferred to stay with your current approach of supporting a single `style.css` to override as it would end up being a breaking change for existing users.

You can do a bit of a silly workaround which is basically the same as what is supported using the method above anyway. Create a `static/assets/tomato.css` file and make your custom changes, but you still need to import a base style:
```css
@import 'orange.css';

 :root {
   --accent: #dd225a;
 }
```

Then in the `config.toml`:
```toml
[params]
  themeColor = "tomato"
```

The only benefit of this is you can remove the need to load in a separate style.css after your custom theme, but it's really a negligible benefit.

That said, the real benefit that this PR does implement is for power users. I mentioned earlier that some of the styles rely on `--color-mod` at build time and so these can't be overridden. Well the benefit of splitting out the base colours means that if someone forks the repo and creates their own palette (for example, I've made my own purple theme by making a new file `source/css/color/purple.css`. Now when I set my variables and build with yarn I get a completely clean palette without any orange styles sneaking into the new build.

### Extending

I see this as the first step in allowing for more customisation. The frustration I had trying to override the base styles using just the `static/style.css` was simply because there was too much that needed an override. A good example of this is the header colour being hardcoded to white when it could be using a variable but there's quite a few that I've been taking note of and I plan to submit them as a separate PR to clean it up a bit. I just wasn't sure how open you would be to some of these things. Your theme is fantastic and I didn't want to sound like I was disapproving of your work!

One of the things I wanted for my blog was to allow automatic light and dark mode switching that follows the OS preference. I've now got this working but it's a lot of overriding. There's opportunities to extend the css variable coverage to both allow more to be overridden in custom palettes, and also to cut down on the amount of overrides someone needs to place in their own stylesheet.

Sorry, this turned into a really long post! If you're still happy for this change let me know and I'll update the PR to match the recent commits on the `master` branch.
